tests/queries/0_stateless/02096_date_time_1970_saturation.sql

```diff
@@ -0,0 +1,31 @@
+select toDate(0);
+select toDateTime(0, 'Europe/Moscow');
+select toMonday(toDate(0));
+select toMonday(toDateTime(0, 'Europe/Moscow'));
+select toStartOfWeek(toDate(0));
+select toStartOfWeek(toDateTime(0, 'Europe/Moscow'));
+select toStartOfMonth(toDate(0));
+select toStartOfMonth(toDateTime(0, 'Europe/Moscow'));
+select toStartOfQuarter(toDate(0));
+select toStartOfQuarter(toDateTime(0, 'Europe/Moscow'));
+select toStartOfYear(toDate(0));
+select toStartOfYear(toDateTime(0, 'Europe/Moscow'));
+select toTime(toDateTime(0, 'Europe/Moscow'));
+select toStartOfMinute(toDateTime(0, 'Europe/Moscow'));
+select toStartOfFiveMinute(toDateTime(0, 'Europe/Moscow'));
+select toStartOfTenMinutes(toDateTime(0, 'Europe/Moscow'));
+select toStartOfFifteenMinutes(toDateTime(0, 'Europe/Moscow'));
+select toStartOfHour(toDateTime(0, 'Europe/Moscow'));
+
+select toDateTime(0, 'America/Los_Angeles');
+select toMonday(toDateTime(0, 'America/Los_Angeles'));
+select toStartOfWeek(toDateTime(0, 'America/Los_Angeles'));
+select toStartOfMonth(toDateTime(0, 'America/Los_Angeles'));
+select toStartOfQuarter(toDateTime(0, 'America/Los_Angeles'));
+select toStartOfYear(toDateTime(0, 'America/Los_Angeles'));
+select toTime(toDateTime(0, 'America/Los_Angeles'), 'America/Los_Angeles');
+select toStartOfMinute(toDateTime(0, 'America/Los_Angeles'));
```

Comment:
@amosbird Any reason that you decided not to saturate `toStartOfMinute` and friends?

Reply:
It should be.

```
┌─toStartOfMinute(toDateTime(0, 'America/Los_Angeles'))─┐
│                                   1969-12-31 16:00:00 │
└───────────────────────────────────────────────────────┘
```

Reply:
Actually it does not always work.

#### `toStartOfHour`

Does not work for partial hours timezones, i.e. `America/Paramaribo`

```sql
┌─toStartOfHour(toDateTime(0, 'America/Paramaribo'))─┐
│                                2106-02-07 02:58:16 │
└────────────────────────────────────────────────────┘
```

#### `toStartOfMinute`

Does not work for partial minutes timezones, `Africa/Monrovia` - which is very special timezone, #27080 for the reference.

```sh
$ TZ=Africa/Monrovia date -d@1
Wed Dec 31 11:15:31 PM MMT 1969
                 ^^
```

```sql
┌─toStartOfMinute(toDateTime(0, 'Africa/Monrovia'))─┐
│                               2106-02-07 06:27:46 │
└───────────────────────────────────────────────────┘
```

And this breaks monotonicity of the function.

It can be fixed to avoid breaking monotonicity, but this will make the result of functions incorrect and also breaks backward compatibility for negative timestamps.